src/server/shttpd/io_emb.c

```diff
@@ -306,7 +306,7 @@ shttpd_wakeup(const void *priv)
 	conn->flags &= ~SHTTPD_SUSPEND;
 #endif
 	(void) memcpy(buf, &cmd, sizeof(cmd));
-	(void) memcpy(buf + sizeof(cmd), conn, sizeof(conn));
+	(void) memcpy(buf + sizeof(cmd), conn, sizeof(struct conn));
```

Comment:
Argh, I should've looked closer before. This is some really ugly piece of code in `shttpd`. Sorry for the confusion.

Problem: The `buf` memcpy is using here is sized to `sizeof(int) + sizeof(void *)`. So it only has space for a _pointer_ (to `struct conn`) but not for the whole struct.
This means that your first fix (adding `&` to the `conn` parameter of `memcpy`) was actually right.

Reply:
Biggest question here: How to test this ? :-/